### DISCUSSION TITLE
Automatic column width for latex equations in html page

### DISCUSSION THREAD

**Discussion by jpmorr** [quarto-dev/quarto-cli#2170] [Q&A] [answered]
Is there a way to specify that an equation should fit within a column width?

I have an equation that is quite long and extends beyond the column width in html format:
![image](https://user-images.githubusercontent.com/10863809/187045521-a1d065f6-fb45-4f40-8b1f-1ab036e9d14f.png)
![image](https://user-images.githubusercontent.com/10863809/187046095-a8975faf-75d7-4e29-afd4-fe67b59e5660.png)

In the latex generated pdf it is properly formatted to the page width as it seems to automatically adjust the font size to make it fit:
![image](https://user-images.githubusercontent.com/10863809/187045537-83bae62a-175f-4602-8ad3-7160450c1593.png)

Here's the equation:
```latex
$$
\{\Delta F\} = \{\Delta F_{\alpha x} \ \Delta F_{\alpha y} \ \Delta F_{\alpha z} \  \Delta M_{\alpha x} \  \Delta M_{\alpha y} \ \Delta M_{\alpha z} \ \Delta F_{\beta x} \  \Delta F_{\beta y} \ \Delta F_{\beta z} \ \Delta M_{\beta x} \  \Delta M_{\beta y} \ \Delta M_{\beta z} \}
$$ {#eq-DeltaF}

```

How do I make the html fit the equation to the column properly?

**Comment by mcanouil** [comment, marked as answer]
This is a MathJax issue (or KaTeX, etc.).
There is an extension to MathJax that seems to do what you want: <https://mathjax.github.io/MathJax-a11y/docs/>.

You can see the following related question to MathJax extension: <https://github.com/quarto-dev/quarto-cli/discussions/1620>.

**Reply by jpmorr** [reply]
@mcanouil I'm not sure that the auto-collapse.js extensions are what I am after - the documentation (and a linked CodePen example) suggest that is changing the display of the equation by altering or factoring the equation or not displaying a part when clicked on. I simply want the equation to render in a slightly smaller font to not extend beyond the column.

I've tried rendering with `katex` and it's both an improvement and worse at the same time. The equation numbers now respect the column width but the equation font size is not modified and now overlays it's own equation number.

![image](https://user-images.githubusercontent.com/10863809/187069497-7cd74313-d512-4e3f-9254-d9d23964d0cb.png)

**Reply by mcanouil** [reply]
Manually you could overwrite the css to change the font-size.
(A JavaScript script could also be used)

I assume, a MathJax extension already exist to do what you want. The extension I listed above is simply a results of a really quick search.

**Reply by jpmorr** [reply]
OK, thanks. I will have a search for an appropriate MathJax extension or possibly some latex _vodoo_ for that specific equation.

**Reply by mcanouil** [reply]
@jpmorr Would the following work for you?
<img width="1624" alt="image" src="https://user-images.githubusercontent.com/8896044/187260058-766b33c6-dff6-4614-b42b-c8feafa7bc5c.png">

<details><summary>See the Quarto document content</summary>
---
format: html
---

# Section

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque dignissim enim non leo lacinia volutpat quis eu felis. Suspendisse in vehicula elit. Nam imperdiet odio sapien, eu congue diam facilisis nec. Vestibulum egestas, tortor eu euismod dictum, nulla diam dapibus augue, sit amet fermentum felis turpis nec risus. Sed ipsum lectus, ultricies vitae lacinia a, vulputate quis lectus. Aliquam mattis est vel neque aliquam, et pellentesque nisi mattis. Pellentesque tristique ultricies tortor, eget pulvinar leo ornare at.

:::{style="font-size:50%;"}
$$
\{\Delta F\} = \{\Delta F_{\alpha x} \ \Delta F_{\alpha y} \ \Delta F_{\alpha z} \  \Delta M_{\alpha x} \  \Delta M_{\alpha y} \ \Delta M_{\alpha z} \ \Delta F_{\beta x} \  \Delta F_{\beta y} \ \Delta F_{\beta z} \ \Delta M_{\beta x} \  \Delta M_{\beta y} \ \Delta M_{\beta z} \}
$$ {#eq-DeltaF}
:::

Etiam dapibus pellentesque nisl, vitae condimentum nisl commodo id. Curabitur odio tellus, auctor ut eleifend ut, bibendum lobortis tellus. Sed eleifend porta lorem id scelerisque. Nunc eu orci turpis. Aliquam suscipit velit in justo lobortis, vel feugiat ligula pulvinar. Fusce sed cursus arcu. Etiam et cursus eros, ac dignissim nibh. Maecenas arcu dui, egestas vitae diam et, imperdiet pharetra eros. Aliquam sagittis purus et est volutpat, quis posuere justo condimentum. Maecenas fermentum nunc quam, non rutrum lorem vestibulum non. Etiam eu finibus arcu, eget consectetur dolor. Maecenas ultricies magna accumsan felis auctor eleifend eu ultricies urna.

</details>

**Reply by jpmorr** [reply]
@mcanouil That's a really nice solution. I tried using `\small` in the latex environment to resize it and that works better for the HTML output as it keeps the equation number the correct size but slightly reduces the actual equation whereas `{style=}` reduce the equation number as well:

![image](https://user-images.githubusercontent.com/10863809/187264227-ac2d2c57-45d1-4728-beb1-a64d1f95734c.png)

(style modification is equation 3.4, \small is 3.5)

However I think for the pdf generation, the style solution is better as it's ignored when creating the pdf and the font size doesn't become too small

**Reply by mcanouil** [reply]
If you want/need a better solution, then I suggest for you to focus your search in a MathJax, KaTeX, etc. and/or LaTeX context rather than Quarto.

For example, one could use horizontal scroll bar:
```
:::{style="overflow-x:auto;overflow-y:hidden;"}
$$
\{\Delta F\} = \{\Delta F_{\alpha x} \ \Delta F_{\alpha y} \ \Delta F_{\alpha z} \  \Delta M_{\alpha x} \  \Delta M_{\alpha y} \ \Delta M_{\alpha z} \ \Delta F_{\beta x} \  \Delta F_{\beta y} \ \Delta F_{\beta z} \ \Delta M_{\beta x} \  \Delta M_{\beta y} \ \Delta M_{\beta z} \}
$$ {#eq-DeltaF}
:::
```

https://user-images.githubusercontent.com/8896044/187270663-464ac3fa-1592-46c4-be8d-bdb7fd2e86a4.mov

**Reply by jpmorr** [reply]
@mcanouil That's pretty cool, but I think overkill in this situation. Both of the previous solutions really work quite well and I think I'd prefer to just show a slightly smaller font than scroll the equation. Thanks for all the great ideas though!

**Reply by mcanouil** [reply]
It was only to show you that with css you can do whatever you want and suit your use case.
The solutions are nearly infinite. 😉
